### PR TITLE
refactor: migrate typescript

### PR DESCRIPTION
**Pull request by yoshinorin**

### CODE REVIEW

**Comment by yoshinorin** [comment]
@hexojs/core 
Maybe ready. I think we can improve more type-safe. But, I want to merge this once before brush-up.

File: lib/log.ts [hunk 1]
```diff
@@ -34,10 +32,31 @@ const console = new Console({
   colorMode: false
 });
 
+type Options = {
+  debug?: boolean,
+  silent?: boolean
+}
```
Comment (renbaoshuo) [inline lib/log.ts:38]:
```suggestion
interface Options {
  debug?: boolean,
  silent?: boolean
}
```

File: lib/log.ts [hunk 2]
```diff
@@ -141,4 +160,4 @@ function createLogger(options) {
   return logger;
 }
 
-module.exports = createLogger;
+export const logger = (option: Options) => createLogger(option);
```
Comment (renbaoshuo) [inline lib/log.ts:163]: I don't think it's suitable here.

Comment (yoshinorin) [reply lib/log.ts:163]: OK, I'll check it.

Comment (yoshinorin) [reply lib/log.ts:163]: @renbaoshuo 
Done. Ready :)

**Review by stevenjoezhang — APPROVED** [review]

**Pull request merged**